.github/workflows/ci.yml

```diff
@@ -0,0 +1,35 @@
+name: Run Tests
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+  pull_request:
+
+jobs:
+  test:
+    name: Test (${{ matrix.os }} - ${{ matrix.arch }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ macOS-latest, ubuntu-latest, windows-latest ]
+        arch: [ x64 ]
+        include:
+          - os: macOS-latest
+            arch: arm64
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Node.js 16.x
+        uses: actions/setup-node@v3.4.1
+        with:
+          node-version: 16.x
+          cache: yarn
+      - name: Install
+        run: yarn install
```

Comment:
Not sure how crucial is this, but the docs [recommend](https://yarnpkg.com/cli/install#options-immutable) (The para below it explains better about --immutable) using [`yarn install --frozen-lockfile`](https://classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-frozen-lockfile)/`yarn install --immutable` when working with CI?

Reply:
@aryanshridhar this is true, we can add that in a separate PR!